src/main/java/duke/command/DeleteCommand.java

```diff
@@ -0,0 +1,51 @@
+package duke.command;
+
+import duke.storage.Storage;
+import duke.tasks.TaskList;
+import duke.ui.Ui;
+
+public class DeleteCommand extends Command {
+    private final int index;
+
+    /**
+     * Initialises a DeleteCommand.
+     *
+     * @param index The index of the task to be deleted.
+     */
+    public DeleteCommand(int index) {
+        this.index = index;
+    }
+
+    /**
+     * Executes the Delete Command and removes the task from the Linked List and file.
+     *
+     * @param tasks   The TaskList containing all existing tasks.
+     * @param ui      The Ui for printing purposes.
+     * @param storage The Storage for saving tasks to file.
+     */
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
+
+        StringBuilder sb = new StringBuilder("Noted. I've removed this task:\n" + tasks.allTasks.get(index));
```

Comment:
Good use of String Buffer to increase efficiency rather than building a new string each time.